types/clamscan/index.d.ts

```diff
@@ -111,88 +30,104 @@ declare class NodeClam {
 }
 
 interface NodeClamFunctions {
-    reset(options?: Options, cb?: () => void): Promise<object> | NodeClamScanError | NodeClamFileError;
+    reset(options?: Options, cb?: () => void): Promise<NodeClam>;
 
-    getVersion(cb?: () => void): Promise<string> | NodeClamScanError | NodeClamFileError;
+    getVersion(cb?: (err: Error | null, version: string) => void): Promise<string>;
 
-    isInfected(file: string, cb?: () => void): Promise<{
+    isInfected(
+      file: string,
+      cb?: (err: Error | null, file: string, isInfected: boolean, viruses: string[]) => void
+    ): Promise<{
         file: string;
         isInfected: boolean;
-        viruses: any[];
+        viruses: string[];
     }>;
 
     passthrough(): {
         isInfected: boolean,
-        viruses: any[]
-    } | NodeClamScanError | NodeClamFileError;
+        viruses: string[]
+    };
 
-    scanFile(file: string,  cb?: () => void): Promise<{
+    scanFile(
+      file: string,
+      cb?: (err: Error | null, file: string, isInfected: boolean, viruses: string[]) => void
+    ): Promise<{
         file: string;
         isInfected: boolean;
-        viruses: any[];
-    }> | NodeClamScanError | NodeClamFileError;
-
-    scanFiles(files: any[], endCb?: () => void, fileCb?: () => void): Promise<{
-        goodFiles: any[];
-        badFiles: any[];
-        errors: object;
-        viruses: any[]
-    }> | NodeClamScanError | NodeClamFileError;
+        viruses: string[];
+    }> ;
+
+    scanFiles(
+      files: string[],
+      endCb?: (err: Error | null, goodFiles: string[], badFiles: string[], viruses: string[]) => void,
+      fileCb?: (err: Error | null, file: string, isInfected: boolean, viruses: string[]) => void
+    ): Promise<{
+        goodFiles: string[];
+        badFiles: string[];
+        errors: {
+            [filename: string]: Error
+        };
+        viruses: string[]
+    }>;
 
-    scanDir(path: string, endCb?: () => void, fileCb?: () => void): Promise<{
+    scanDir(
+      path: string,
+      endCb?: (err: Error | null, goodFiles: string[], badFiles: string[], viruses: string[]) => void,
+      fileCb?: (err: Error | null, file: string, isInfected: boolean, viruses: string[]) => void
+    ): Promise<{
         path: string;
         isInfected: boolean;
-        goodFiles: any[];
-        badFiles: any[];
-        viruses: any[];
-    }> | NodeClamScanError | NodeClamFileError;
+        goodFiles: string[];
+        badFiles: string[];
+        viruses: string[];
+    }>;
 
-    scanStream(stream: ReadableStream, cb?: () => void): Promise<{
+    scanStream(stream: ReadableStream, cb?: (err: Error | null, isInfected: boolean) => void): Promise<{
```

Comment:
```
    /**
     * This method allows you to scan a binary stream. NOTE: This method will
     * only work if you've configured the module to allow the use of a TCP or
     * UNIX Domain socket. In other words, this will not work if you only have
     * access to a local ClamAV binary.
     *
     * @see {@link https://github.com/kylefarris/clamscan#scanstreamstreamcallback}
     */
    scanStream(stream: ReadableStream): Promise<{
        file: string;
        isInfected: boolean;
        viruses: string[];
    }>;
    scanStream(stream: ReadableStream, cb?: (err: Error | null, isInfected: boolean) => void): void;
```